extensions/espanso/src/index.tsx

```diff
@@ -43,32 +75,138 @@ export default function Command() {
     fetchData();
   }, []);
 
+  useEffect(() => {
+    if (selectedCategory === "All") {
+      setFilteredItems(items);
+    } else {
+      setFilteredItems(
+        items.filter(
+          (item) => (item.label?.includes(" > ") ? item.label.split(" > ")[0] : "Miscellaneous") === selectedCategory,
+        ),
+      );
+    }
+  }, [selectedCategory, items]);
+
   if (error) {
     const notFound = Boolean(/command not found/.exec(error.stderr));
-
     return notFound ? <Detail markdown={commandNotFoundMd} /> : <Detail markdown={error.stderr} />;
   }
 
   if (!isLoading && items.length === 0) {
     return <Detail markdown={noContentMd} />;
   }
 
+  const groupBySections = (matches: NormalizedEspansoMatch[]) => {
+    return matches.reduce(
+      (sections, match) => {
+        const { label } = match;
+        const sectionKey = label?.includes(" > ") ? label.split(" > ")[0] : "Miscellaneous";
+        if (!sections[sectionKey]) {
+          sections[sectionKey] = [];
+        }
+        sections[sectionKey].push(match);
+        return sections;
+      },
+      {} as Record<string, NormalizedEspansoMatch[]>,
+    );
+  };
+
+  const sections = groupBySections(filteredItems);
+
+  const sortedSectionKeys = Object.keys(sections).sort((a, b) => {
+    if (a === "Miscellaneous") return 1;
+    if (b === "Miscellaneous") return -1;
+    return a.localeCompare(b);
+  });
+
   return (
-    <List isShowingDetail isLoading={isLoading}>
-      {items.map(({ triggers, replace, label }, index) => (
-        <List.Item
-          key={index}
-          title={label ?? triggers.join(", ")}
-          subtitle={!label ? "" : triggers.join(", ")}
-          detail={<List.Item.Detail markdown={replace} />}
-          actions={
-            <ActionPanel>
-              <Action title={pasteTitle} onAction={() => Clipboard.paste(replace)} />
-              <Action.CopyToClipboard title="Copy Content" content={replace} />
-              <Action.CopyToClipboard title="Copy Triggers" content={triggers.join(", ")} />
-            </ActionPanel>
-          }
-        />
+    <List
+      isShowingDetail
+      isLoading={isLoading}
+      searchBarAccessory={<CategoryDropdown categories={categories} onCategoryChange={setSelectedCategory} />}
+    >
+      {sortedSectionKeys.map((sectionKey) => (
+        <List.Section key={sectionKey} title={sectionKey}>
+          {sections[sectionKey].map(({ triggers, replace, label, filePath }, index) => {
+            let title = triggers.join(", ");
+            let subtitle: string | undefined = undefined;
+
+            let category: string | undefined = undefined;
+            let subcategory: string | undefined = undefined;
+            let remainingLabel: string | undefined = undefined;
+
+            if (label) {
+              const parts = label.split(" > ");
```

Comment:
Since the `label.split(" > ")` is duplicated quite a few times, maybe create a dedicated function for this logic?